hooks/useCollectionData.ts

```diff
@@ -0,0 +1,24 @@
+import { getAddress } from "viem";
+import { useMemo } from "react";
+import { useCollectionProvider } from "@/providers/CollectionProvider";
+import { useUserProvider } from "@/providers/UserProvider";
+
+const useCollectionData = () => {
+  const { data: collection } = useCollectionProvider();
+  const { artistWallet } = useUserProvider();
+
+  const isOwner = useMemo(() => {
+    if (!collection || !artistWallet) return false;
+    const defaultAdminAddress = collection.default_admin?.address;
+    if (!defaultAdminAddress) return false;
+    return getAddress(artistWallet) === getAddress(defaultAdminAddress);
+  }, [collection, artistWallet]);
+
+  return {
+    isOwner,
+    metadata: collection?.metadata ?? null,
+    isLoading: !collection,
+  };
```

Comment:
DRY principle - Why are we exporting `provider` variables from another hook?
- actual: returning `metadata` and `isLoading` from a collection which could be accessed directly via `useCollectionProvider`
- required: rename this hook to `useIsOwner` and only return the `isOwner` variable.